File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/CreditCard/Enrollment/Reaction/EnrollmentReactionController.cs
using CreditCardEnrollment.Common.Ambar;
using CreditCardEnrollment.Common.Reaction;
using CreditCardEnrollment.Common.SerializedEvent;
using CreditCardEnrollment.Common.EventStore;
using CreditCardEnrollment.Common.Projection;
using Microsoft.AspNetCore.Mvc;

namespace CreditCardEnrollment.CreditCard.Enrollment.Reaction;

[ApiController]
[Route("api/v1/credit_card/enrollment/reaction")]
[Produces("application/json")]
[Consumes("application/json")]
public class EnrollmentReactionController : ReactionController {
    private readonly ReviewEnrollmentReactionHandler _reviewEnrollmentReactionHandler;

    public EnrollmentReactionController(
        PostgresTransactionalEventStore eventStore,
        MongoTransactionalProjectionOperator mongoOperator,
        Deserializer deserializer,
        ReviewEnrollmentReactionHandler reviewEnrollmentReactionHandler)
        : base(eventStore, mongoOperator, deserializer) {
        _reviewEnrollmentReactionHandler = reviewEnrollmentReactionHandler;
    }

    [HttpPost("review_enrollment")]
    public IActionResult ReactWithReviewEnrollment([FromBody] AmbarHttpRequest request) {
        return new ContentResult {
            Content = ProcessReactionHttpRequest(
                request, 
                _reviewEnrollmentReactionHandler
            ),
            ContentType = "application/json",
            StatusCode = 200
        };
    }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/creditCard/enrollment/reaction/EnrollmentReactionController.ts
import { ReactionController } from '../../../common/reaction/ReactionController';
import { AmbarHttpRequest } from '../../../common/ambar/AmbarHttpRequest';
import { ReviewEnrollmentReactionHandler } from './ReviewEnrollmentReactionHandler';
import { PostgresTransactionalEventStore } from '../../../common/eventStore/PostgresTransactionalEventStore';
import { MongoTransactionalProjectionOperator } from '../../../common/projection/MongoTransactionalProjectionOperator';
import { Deserializer } from '../../../common/serializedEvent/Deserializer';
import {Request, Response, Router} from "express";
import {inject, injectable} from "tsyringe";

@injectable()
export class EnrollmentReactionController extends ReactionController {
    public readonly router: Router;

    constructor(
        @inject(PostgresTransactionalEventStore) eventStore: PostgresTransactionalEventStore,
        @inject(MongoTransactionalProjectionOperator) mongoOperator: MongoTransactionalProjectionOperator,
        @inject(Deserializer) deserializer: Deserializer,
        @inject(ReviewEnrollmentReactionHandler) private readonly reviewEnrollmentReactionHandler: ReviewEnrollmentReactionHandler
    ) {
        super(eventStore, mongoOperator, deserializer);
        this.router = Router();
        this.router.post('/review_enrollment', this.reactWithReviewEnrollment.bind(this));
    }

    async reactWithReviewEnrollment(req: Request, res: Response): Promise<void> {
        const response = await this.processReactionHttpRequest(
            req.body as AmbarHttpRequest,
            this.reviewEnrollmentReactionHandler,
        );
        res.status(200).contentType('application/json').send(response);
    }
}
